### PR TITLE
With_demo_project

### LINKED ISSUE #21: Add examples to all functions
Requires #8.

### LINKED ISSUE #120: Create with_example_project()?
Via https://github.com/cynkra/fledge/pull/113#discussion_r708335953

> Combine `create_demo_project()` and `with_project()`. I'm not sure it's a good idea though.

### PR DESCRIPTION
**Pull request by maelle**
Fix #120 
Fix #21 

I put examples in R scripts this way it's easier to write, indent and run them.

### CODE REVIEW

File: R/api-bump-version.R [hunk 1]
```diff
@@ -14,27 +14,7 @@
...
-#' # Pretend we added useful code inside it.
-#' # Track the new R file with Git.
-#' gert::git_add("R/cool-function.R")
-#' gert::git_commit("- Add cool function.")
-#' # Bump version with fledge.
-#' fledge::bump_version()
-#' })
-#' # Clean
-#' unlink(tempdir, recursive = TRUE)
+#' @example man/examples/bump-version.R
```
Comment (maelle) [inline R/api-bump-version.R:17]: Putting examples in R scripts to

- indent them
- run them
- reuse them

File: R/repo-helper.R [hunk 2]
```diff
@@ -91,3 +91,37 @@ set_usethis_desc <- function(maintainer, email, date) {
...
   )
 }
+
+
+#' Run code in temporary project
+#'
+#' @inheritParams create_demo_project
+#' @param code Code to run with temporary active project
+#'
+#' @details If you open the project you are responsible for deleting it.
```
Comment (maelle) [inline R/repo-helper.R:101]: How does this sound?
Otherwise once one gets to the project no file can be opened.

Comment (maelle) [reply R/repo-helper.R:101]: or maybe we can remove the open argument.

File: man/examples/commit-version.R [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  # (Or go and actually run code inside the newly created package project!)
+  # Add a new R file.
+  usethis::use_r("cool-function", open = FALSE)
+  # Pretend we added useful code inside it.
+  # Track the new R file with Git.
+  gert::git_add("R/cool-function.R")
+  gert::git_commit("- Add cool function.")
+  # Bump version with fledge.
+  fledge::bump_version()
+  # Edit `NEWS.md`, convert the "change log" to "release notes" --
```
Comment (maelle) [inline man/examples/commit-version.R:14]: Does this example look ok?

File: man/examples/commit-version.R [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  # Add a new R file.
+  usethis::use_r("cool-function", open = FALSE)
+  # Pretend we added useful code inside it.
+  # Track the new R file with Git.
+  gert::git_add("R/cool-function.R")
+  gert::git_commit("- Add cool function.")
+  # Bump version with fledge.
+  fledge::bump_version()
+  desc::desc_add_author(given = "Jane", family = "Doe", role = "ctb")
+  fledge::commit_version()
```
Comment (maelle) [inline man/examples/commit-version.R:15]: does this make any sense?

**Comment by maelle** [comment]
in the pkgdown preview one sees examples rendered :sparkles:

File: man/examples/tag-version.R [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  # (Or go and actually run code inside the newly created package project!)
+  # Add a new R file.
+  usethis::use_r("cool-function", open = FALSE)
+  # Pretend we added useful code inside it.
+  # Track the new R file with Git.
+  gert::git_add("R/cool-function.R")
+  gert::git_commit("- Add cool function.")
+  # Bump version with fledge.
+  fledge::bump_version()
+  fledge::update_news(c("- something I forgot", "- blabla"))
```
[the remark this answers is not in the document]
Comment (maelle) [reply man/examples/tag-version.R:14]: it does demonstrate `update_news()`, `update_version()` and `tag_version()` though

File: R/repo-helper.R (hunk 2 above)
Comment (krlmlr) [reply R/repo-helper.R:101]: Let's remove `open`, we have `create_demo_project()` that exposes it. (If it's not exported, we could.)

**Review by krlmlr — COMMENTED** [review]
Thanks, looks great!

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**